### PR TITLE
Project: delete the `.jobsets` jobset if project is no longer declarative

### PR DESCRIPTION
**Pull request by cole-h**
"No longer declarative" as defined by the "Edit project" page is an empty spec
file.

---

Fixes https://github.com/NixOS/hydra/issues/960.

~~TODO: write a test~~

### CODE REVIEW

File: src/lib/Hydra/Controller/Project.pm [hunk 1]
```diff
@@ -169,6 +169,8 @@ sub updateProject {
             , emailoverride => ""
             , triggertime => time
             });
+    } else {
+        $project->jobsets->search({ name => ".jobsets" })->delete;
```
Comment (cole-h) [inline src/lib/Hydra/Controller/Project.pm:173]: Should we maybe also clear the `decltype` and `declvalue`?

Comment (cole-h) [reply src/lib/Hydra/Controller/Project.pm:173]: I opted to do so, but it can be easily reverted.

**Comment by cole-h** [comment]
Ready for review, but NOT ready for merge (if clearing `decltype` and `declvalue` are deemed acceptable when going back to a normal jobset, 98633e75d65aad71e8b9621b6e9cb5e7a1cffc11 will need to be reworded to remove the question mark).

File: src/lib/Hydra/Controller/Project.pm (hunk 1 above)
Comment (grahamc) [reply src/lib/Hydra/Controller/Project.pm:173]: I think this is a good idea.

File: t/Controller/projects.t [hunk 2]
```diff
@@ -17,27 +17,133 @@ Catalyst::Test->import('Hydra');
...
+    Content => {
+        username => 'alice',
+        password => 'foobar'
     }
-});
+);
+is($req->code, 302);
+my $cookie = $req->header("set-cookie");
+
+subtest "Read project 'tests'" => sub {
```
Comment (grahamc) [inline t/Controller/projects.t:38]: Thanks for moving these in to subtests.

**Review by grahamc — APPROVED** [review]
Confirmed it works locally on a declarative project which has no generated projects. I then created a declarative project which did create jobsets, emptied the file field, saved, and the `.jobsets` jobset was removed and other jobsets remained.

**Pull request merged**